src/rust/ide/src/ide.rs

```diff
@@ -0,0 +1,184 @@
+    }
+
+    /// Wraps the transport to the project manager server into the client type and registers it
+    /// within the global executor.
+    pub fn setup_project_manager
+    (transport:impl json_rpc::Transport + 'static) -> project_manager::Client {
+        let project_manager = project_manager::Client::new(transport);
+        executor::global::spawn(project_manager.runner());
+        project_manager
+    }
+
+    /// Creates a new websocket transport and waits until the connection is properly opened.
+    pub async fn new_opened_ws
+    (logger:Logger, address:project_manager::IpWithSocket) -> Result<WebSocket,ConnectingError> {
+        let endpoint = format!("ws://{}:{}", address.host, address.port);
+        WebSocket::new_opened(logger,endpoint).await
+    }
+
+    /// Connect to language server.
+    pub async fn open_project
+    ( logger          : &Logger
+      , json_endpoint   : project_manager::IpWithSocket
+      , binary_endpoint : project_manager::IpWithSocket
+      , project_name    : impl Str
+    ) -> FallibleResult<controller::Project> {
+        info!(logger, "Establishing Language Server connections.");
+        let client_id     = Uuid::new_v4();
+        let json_ws       = Self::new_opened_ws(logger.clone_ref(), json_endpoint).await?;
+        let binary_ws     = Self::new_opened_ws(logger.clone_ref(), binary_endpoint).await?;
+        let client_json   = language_server::Client::new(json_ws);
+        let client_binary = binary::Client::new(logger,binary_ws);
+        crate::executor::global::spawn(client_json.runner());
+        crate::executor::global::spawn(client_binary.runner());
+        let connection_json   = language_server::Connection::new(client_json,client_id).await?;
+        let connection_binary = binary::Connection::new(client_binary,client_id).await?;
+        Ok(controller::Project::new(logger,connection_json,connection_binary,project_name))
+    }
+
+    /// Creates a new project and returns its metadata, so the newly connected project can be
+    /// opened.
+    pub async fn create_project
+    ( logger          : &Logger
+    , project_manager : &impl project_manager::API
+    , name            : &String) -> FallibleResult<ProjectMetadata> {
+        info!(logger, "Creating a new project named `{name}`.");
+        let id = project_manager.create_project(name).await?.project_id;
+        Ok(ProjectMetadata {
+            id,
+            name        : ProjectName {name:name.clone()},
+            last_opened : None,
+        })
+    }
+
+    /// Open the named project or create a new project if it doesn't exist.
+    pub async fn open_project_or_create_new
+    ( logger : &Logger
+    , project_manager : &impl project_manager::API
+    , project_name    : &String) -> FallibleResult<controller::Project> {
+        let projects_to_list = constants::MAXIMUM_LISTABLE_PROJECTS;
+        let response     = project_manager.list_recent_projects(&projects_to_list).await?;
```

Comment:
This should be noted with a TODO note that it should be changed once Engine adjusts their API as planned. 
See https://github.com/luna/enso/issues/819